### PR TITLE
[do not merge] proof-of-concept sticky content changing

### PR DESCRIPTION
**Pull request by ds300**
changing-sticky-content

### CODE REVIEW

File: src/lib/Containers/Artist.tsx [hunk 1]
```diff
@@ -81,14 +84,56 @@ export const Artist: React.FC<{
...
   )
 }
 
+const ArtistStickyHeader: React.FC<{}> = ({}) => {
+  const { headerOffsetY, staticHeaderHeight } = useStickyTabPageContext()
+  const _isHeaderStuck = useMemo(() => {
+    return Animated.eq(headerOffsetY, Animated.multiply(-1, staticHeaderHeight ?? +9999999))
+  }, [headerOffsetY, staticHeaderHeight])
+
+  const isHeaderStuck = Boolean(useNativeValue(_isHeaderStuck, 0))
```
Comment (ds300) [inline src/lib/Containers/Artist.tsx:105]: the sticky tab page is built on these so-called 'Animated' values, which are abstract wrappers for native values. Here we create a new native value to represent whether or not the header is in it's 'stuck' position, and then deref the native value using this special hook `useNativeValue`

File: src/lib/Containers/Artist.tsx [hunk 2]
```diff
@@ -81,14 +84,56 @@ export const Artist: React.FC<{
...
+
+  const overlayOpacity = useRef(new Animated.Value(0)).current
+
+  useEffect(() => {
+    Animated.timing(overlayOpacity, { toValue: isHeaderStuck ? 1 : 0, duration: 140, easing: Easing.ease }).start()
+  }, [isHeaderStuck])
```
Comment (ds300) [inline src/lib/Containers/Artist.tsx:111]: then we feed that `isHeaderStuck` boolean into some animation code to control the opacity of the  overlay content

**Comment by ds300** [comment]
gonna close this to clean up

**Comment by MounirDhahri** [comment]
Nice initiative @ds300 !

**Comment by damassi** [comment]
(Yah @ds300 - it will be a sec before this gets implemented.)

**Pull request closed without merging**